implementations/rust/ockam/ockam_api/src/util.rs

```diff
@@ -308,6 +319,20 @@ pub fn is_local_node(ma: &MultiAddr) -> anyhow::Result<bool> {
     }
 }
 
+/// Tells whether the input [`Code`] references a local worker.
+pub fn local_worker(code: &Code) -> Result<bool> {
+    match *code {
+        Node::CODE | Space::CODE | Project::CODE | DnsAddr::CODE | Ip4::CODE | Ip6::CODE => {
+            Ok(false)
+        }
+        Secure::CODE | Worker::CODE | Service::CODE => Ok(true),
```

Comment:
Yes, the reasoning was that `secure` never changes the scope to another node by itself, but it's always used in conjunction with `tcp` or other transports. That said, I can't find any reasonable use case to create a secure channel within the node.

Reply:
It's probably better to just consider secure remote, since there are many scenarios where this makes sense and zero scenarios where reverse makes sense.